### ISSUE TITLE
Hidden layers are not being hidden when using named maps

### ISSUE THREAD

**Issue by alonsogarciapablo** [CartoDB/Windshaft-cartodb#524]
While investigating https://github.com/CartoDB/cartodb/issues/7918, I realised that Windshaft-cartodb no longer seems to be hiding layers for named maps (in plain english: The HACK doesn't seem to be working now, perhaps it's time to change it).

For example, one of my maps is sending the following payload to the Maps API:
```
{
  "layer0": 0,
  "styles": {}
}
```

And layer is still being rendered. I tried sending `'0'` instead of `0`, and it doesn't make any difference.

cc: @rochoa

**Comment by rochoa** [comment]
If the named map does not include a `WHERE 1 = <%= layer0 %>` template item in the SQL the `layer0` value won't be used to _hide_ the layer.

Closing as this is not a bug in Maps API.
